### ISSUE TITLE
Eureka 1.x - there is no mechanism provided to remove overridden status

### ISSUE THREAD

**Issue by tbak** [Netflix/eureka#389]
Eureka 1 allows one to override an instance status via issuing the following REST call:
```PUT /eureka/v2/apps/appID/instanceID/status?value=OUT_OF_SERVICE```
This is useful if we want to take otherwise healthy instance out of service. The instance info data contain afterwards two status related fields:
```
<status>OUT_OF_SERVICE</status>
<overriddenstatus>OUT_OF_SERVICE</overriddenstatus>
```
To mark instance as up we can modify again the overriddenstatus field by calling:
```PUT /eureka/v2/apps/appID/instanceID/status?value=UP```
If however the instance real state change(s), that is hidden by the enforced value.

We can address this issue in two ways:
* support DELETE HTTP method to remove the overridden status
* merge the two states (see table below)

Status | Overridden status | Merge
---------|-------------------------|---------
DOWN | DOWN | DOWN
DOWN | * | DOWN
UP | OUT_OF_SERVICE | OUT_OF_SERVICE
UP | UP | UP

**Comment by NiteshKant** [comment]
I would prefer the support for DELETE overridden status for the following reason:

The merge algorithm addresses the "most common" usecase for today (for Netflix). I can think of corner cases when someone wants to override DOWN status from the instance with UP, may be in the case the healthcheck has an issue or it is flapping. 

There is value in being consistent with the definition of "overridden status" that it __always__ overrides the status with no special cases. If the override does not apply anymore, then it should be deleted.

**Comment by tbak** [comment]
This issue is a duplicate of #89.

**Comment by tbak** [comment]
Fixed by #412.
Overridden status can be removed via DELETE REST call.
